### PR TITLE
#164894009 Implement users can add image to article feature

### PR DESCRIPTION
**Pull request by BukkyOmo**
#### Description
Implement upload image while the user is creating an article.

#### Type of change
- [x] Task (non-breaking change)
 

#### Checklist
- [x] Install cloudinary, multer, and multer-storage-cloudinary.
- [x] Configure application with cloudinary by using key and secret from cloudinary.
- [x] Include image in request body.
- [x] Update migration and model to include image.
- [x] Test with postman.

#### Pivotal Tracker
[#164894009](https://www.pivotaltracker.com/story/show/164894009)

####Screenshot
Screenshot 2019-04-09 at 4 27 32 AM

### CODE REVIEW

File: server/utils/articles/base.util.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ class baseUtils {
   }
 
   static isBoolean(value) {
-    if (typeof value !== 'boolean') return false;
+    if (typeof value !== 'boolean' && value !== true && value !== false && value !== 'true' && value !== 'false') return false;
```
Comment (geekdanitek) [inline server/utils/articles/base.util.js:38]: Please i would like to seek clarity on this IF statement here.

Comment (BukkyOmo) [reply server/utils/articles/base.util.js:38]: > Please i would like to seek clarity on this IF statement here.

So for you to test an image on postman, you need to use form-data which takes in input as strings but we validated published to be boolean so i had to cater for when it is not true or false and also had to coerce the truthy and falsy values so they can be used in the form-data.

File: server/config/cloudinaryconfig.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+dotenv.config();
+
+cloudinary.config({
+  cloudinary_name: process.env.CLOUDINARY_URL,
+});
+
+const storage = cloudinaryStorage({
+  cloudinary,
+  folder: 'forsetti',
```
Comment (davidokonji) [inline server/config/cloudinaryconfig.js:16]: can you make the folder action specific? i.e there should be a folder specifically for article images, so there can also be another for user images

File: server/controllers/article.controller.js [hunk 3]
```diff
@@ -28,6 +29,7 @@ class ArticleController {
       title,
       body,
       tags,
+      image: [image],
```
Comment (davidokonji) [inline server/controllers/article.controller.js]: Good work, in cases where there are no images in the article, what would happen to this input?

**Review by davidokonji — COMMENTED** [review]
Good work

File: server/config/cloudinaryconfig.js (hunk 2 above)
Comment (BukkyOmo) [reply server/config/cloudinaryconfig.js:16]: > can you make the folder action specific? i.e there should be a folder specifically for article images, so there can also be another for user images

No problem.

File: server/controllers/article.controller.js (hunk 3 above)
Comment (BukkyOmo) [reply server/controllers/article.controller.js]: It returns null

**Comment by TheDrizzyWay** [comment]
Looks good. I would like to know why the image string is in an array.

File: server/config/cloudinaryconfig.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  transformation: [{
+    width: 500, height: 500, crop: 'limit',
+  }],
+});
+
+
+const upload = multer({
+  storage,
+  limits: {
+    fileSize: 2000000,
```
Comment (samorahno) [inline server/config/cloudinaryconfig.js]: Good job @BukkyOmo, I will like to verify what size exactly this is. I will like to suggest if you can make the maximum file size to to be 1mb. The reason for this is to reduce server load and upload time.

**Review by samorahno — COMMENTED** [review]
Good job @BukkyOmo

**Comment by Chiazokam** [comment]
Hi @BukkyOmo looks like your PR title is not in the right format

**Comment by BukkyOmo** [comment]
> Hi @BukkyOmo looks like your PR title is not in the right format

Okay, will definitely edit.

**Review by chukwudidu — COMMENTED** [review]
Looks good to me.

**Pull request merged**